src/gate/index.js

```diff
@@ -54,21 +57,46 @@ Object.assign(UserGate.prototype, {
     return this._matchesSample(user) || this._matchesList(user);
   },
 
+  allowsUniform: function(user) {
+    return (this._matchesUniformSample(user) || this._matchesList(user));
+  },
+
   _matchesList: function(user) {
     if (!this._list) return false;
 
     return this._list.contains(user);
   },
 
   _matchesSample: function(user) {
-    if (!this._sample) return false;
+    if (!this._sample || !((this._sample >= 0) && (this._sample <= 1))) return false;
 
     // See if the user begins with a character in the sample set.
-    var effectiveCharacterSet = this._sampleCharacterSet.slice(
+    const effectiveCharacterSet = this._sampleCharacterSet.slice(
       0, Math.round(this._sampleCharacterSet.length * this._sample));
 
-    var userMatches = new RegExp('^[' + effectiveCharacterSet + ']', 'i').test(user);
+    const userMatches = new RegExp('^[' + effectiveCharacterSet + ']', 'i').test(user);
     return userMatches;
+  },
+
+  _matchesUniformSample: function(user) {
+    if (!this._sample || !((this._sample >= 0) && (this._sample <= 1))) return false;
```

Comment:
Is the first case intended to guard against `this._sample` being undefined, or `this._sample` being zero, or both?

Reply:
Ah, good catch - `this._sample` should be allowed to be 0, so I'll change the check so it's more explicit in its check for false-y values.

Reply:
https://github.com/mixmaxhq/user-gate/pull/5/commits/7e6dfa937363205c20409112b4cd4f9e5553baf0